app/commands/v2/patch_link_set.rb

```diff
@@ -18,17 +18,25 @@ def call
         lock_version.save!
 
         grouped_links.each do |group, payload_content_ids|
+          # For each set of links in a LinkSet scoped by link_type, this iterator
+          # deletes the entire existing set and then imports all the links in the
+          # payload, preserving their ordering. The activerecord-import gem is
+          # used to create a single SQL statement for importing all links.
           links = link_set.links.where(link_type: group)
           existing_content_ids = links.pluck(:target_content_id)
 
-          content_ids_to_create = payload_content_ids - existing_content_ids
-          content_ids_to_delete = existing_content_ids - payload_content_ids
+          links.where(target_content_id: existing_content_ids).delete_all
```

Comment:
@MatMoore We (Publishing API) persuaded Ruben to drop the single insert to multiple ones so that's our fault 😉  We felt it was best to wait until there was performance issues.

@rubenarakelyan I've had a look at the above and I think you can save some a query on each group iteration by dropping the need to get `existing_content_ids`:

```ruby
          link_set.where(link_type: group).delete_all
          payload_content_ids.uniq.each_with_index do |content_id, i|
            links.create!(link_type: group, target_content_id: content_id, position: i)
          end
```

I also dug in a little to see if it would be difficult to go down the update and no-op route if nothing has changed and think this isn't too complex:

```ruby
          links = link_set.links.where(link_type: group)
          existing_content_ids = links.pluck(:target_content_id)
          payload_content_ids.uniq.each_with_index do |content_id, position|
            next unless content_id == existing_content_ids[postion]
            links.find_or_create_by!(target_content_id: content_id).update!(position: position)
          end
          to_delete = existing_content_ids - payload_content_ids
          links.where(target_content_id: to_delete).delete_all unless to_delete.empty?
```